nodeinfo/manager.go

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) 2022 IoTeX Foundation
+// This source code is provided 'as is' and no warranties are given as to title or non-infringement, merchantability
+// or fitness for purpose and, to the extent permitted by law, all liability for your use of the code is disclaimed.
+// This source code is governed by Apache License 2.0 that can be found in the LICENSE file.
+
+package nodeinfo
+
+import (
+	"context"
+	"time"
+
+	"github.com/iotexproject/go-pkgs/cache/lru"
+	"github.com/iotexproject/go-pkgs/crypto"
+	"github.com/iotexproject/go-pkgs/hash"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/iotexproject/iotex-core/pkg/log"
+	"github.com/iotexproject/iotex-core/pkg/routine"
+	"github.com/iotexproject/iotex-core/pkg/util/byteutil"
+	"github.com/iotexproject/iotex-core/pkg/version"
+)
+
+type (
+	transmitter interface {
+		BroadcastOutbound(context.Context, proto.Message) error
+		UnicastOutbound(context.Context, peer.AddrInfo, proto.Message) error
+		Info() (peer.AddrInfo, error)
+	}
+
+	chain interface {
+		TipHeight() uint64
+	}
+
+	// Info node infomation
+	Info struct {
+		Version   string
+		Height    uint64
+		Timestamp time.Time
+		Address   string
+		PeerID    string
+	}
+
+	// InfoManager manage delegate node info
+	InfoManager struct {
+		nodeMap *lru.Cache
+		version string
+		address string
+
+		broadcastTask *routine.RecurringTask
+		transmitter   transmitter
+		chain         chain
+		privKey       crypto.PrivateKey
+	}
+)
+
+var _nodeInfoHeightGauge = prometheus.NewGaugeVec(
+	prometheus.GaugeOpts{
+		Name: "iotex_node_info_height_gauge",
+		Help: "height info of node",
+	},
+	[]string{"address", "version"},
+)
+
+func init() {
+	prometheus.MustRegister(_nodeInfoHeightGauge)
+}
+
+// NewInfoManager new info manager
+func NewInfoManager(cfg *Config, t transmitter, h chain, privKey crypto.PrivateKey) *InfoManager {
+	dm := &InfoManager{
+		nodeMap:     lru.New(_nodeMapSize),
+		transmitter: t,
+		chain:       h,
+		privKey:     privKey,
+		version:     version.PackageVersion,
+		address:     privKey.PublicKey().Address().String(),
+	}
+
+	dm.broadcastTask = routine.NewRecurringTask(func() {
```

Comment:
broadcastTask could be `nil` when `EnableBroadcastNodeInfo` is false

Reply:
remain the task has an advantage that a new delegate can enable broadcast automatically without restart